### ISSUE TITLE
`No such file or directory` error when deleting an existing file

### ISSUE THREAD

**Issue by mlqn** [Altinn/altinn-platform#824]
### Description of the bug

A user from Mattilsynet has a problem with his repository in Studio. The `Hent endringer` or `Slett lokale endringer` buttons dot not work and return a 500 error.

Here is the error logged in Application Insights:
`could not find '/AltinnCore/Repos/Tor_Inge_Sundal/mat/mattilsynet-semintekniker/App/models/model.schema.json' to remove: No such file or directory`

We also get the same error when we try to remove the file manually using `rm -f /AltinnCore/Repos/Tor_Inge_Sundal/mat/mattilsynet-semintekniker/App/models/model.schema.json`.

We have also tried several other techniques (e.g. removing the file using its inum), checked for hidden or special characters in the file name, modified the file attributes, and recreated the pod, but nothing worked.

This is strange as the file is still visible when we run `ls` 🤷‍♂️

### Steps To Reproduce

- run `rm -rf /AltinnCore/Repos/Tor_Inge_Sundal/mat/mattilsynet-semintekniker/App/models/model.schema.json` directly on the disk

**Comment by tjololo** [comment]
Seems to be a problem in azure. When I navigate to the file in the portal the _File properties_ reports that the file is marked for deletion by an SMB client. I will create a support ticket with Microsoft (MS).
Is it a problem that this file is kept until MS has time to investigate or should we try additional steps now to delete the file now so that there is limited possibilities for MS to investigate the root cause of this problem?

**Comment by mlqn** [comment]
We asked the user to change his username, so I believe it's okay that we keep this file until MS has investgated the issue.
Here is the slack thread: https://altinn.slack.com/archives/C02EENBV0Q7/p1692096759202329

**Comment by tjololo** [comment]
I have created a support ticket and will report back as soon as I get an answer from MS

**Comment by tjololo** [comment]
There was an dangling/stuck file handle registered on the file share in azure. After some help from azure support I found the file handle and closed it using the cli. `az storage share close-handle --account-name storage-account-name --name share-name --path "Tor_Inge_Sundal/mat/mattilsynet-semintekniker/App/models" --recursive --close-all`
Had to use close-all on parent directory as pinpointing the file directly resulted in an error.
File and folders are now deleted from the fileshare and I have verified that the file is gone in the designer pod as well.
Closing the issue as resolved.
